apps/users/admin/profile.py

```diff
@@ -1,10 +1,22 @@
 from django.contrib import admin
+from django.db.models.query import QuerySet
 
+from apps.users.factory import get_newsletter_service
 from apps.users.models import Profile
 
+newsletter_service = get_newsletter_service()
+
 
 @admin.register(Profile)
 class ProfileAdmin(admin.ModelAdmin):
     list_display = ("username", "key_access")
 
     search_fields = ["user__username"]
+
+    actions = ["sync_newsletter_subscriptions"]
+
+    def sync_newsletter_subscriptions(self, request, queryset: QuerySet):
+        # TODO hey Bart pls convert to celery shared_task alsjeblieft
```

Comment:
@bcvandendool help s'il vous plait :3

Reply:
Its a shared task now. Please do check if everything still is working correctly, havent had the time to delve deep into testing it.